pkg/commands/os.go

```diff
@@ -126,24 +100,25 @@ func (c *OSCommand) SublimeOpenFile(filename string) (*exec.Cmd, error) {
 }
 
 // OpenFile opens a file with the given
-func (c *OSCommand) OpenFile(filename string) (*exec.Cmd, error) {
-	cmdName, cmdTrail, err := c.GetOpenCommand()
+func (c *OSCommand) OpenFile(filename string) error {
+	cmdName, cmdTrail, err := c.getOpenCommand()
 	if err != nil {
-		return nil, err
+		return err
 	}
-	err = c.RunCommand(cmdName + " " + c.Quote(filename) + cmdTrail) // TODO: test on linux
-	return nil, err
+
+	return c.RunCommand(cmdName + " " + c.Quote(filename) + cmdTrail) // TODO: test on linux
 }
 
 // EditFile opens a file in a subprocess using whatever editor is available,
 // falling back to core.editor, VISUAL, EDITOR, then vi
 func (c *OSCommand) EditFile(filename string) (*exec.Cmd, error) {
-	editor, _ := gitconfig.Global("core.editor")
+	editor, _ := c.getGlobalGitConfig("core.editor")
```

Comment:
I hadn't realised originally that I was adding a git-specific concept to the OS package by using getGlobalGitConfig. Now I see that we are doing that and I would prefer if there was no concept of git in the OS package, however I think it's okay to add an exception in this case, because editing a file isn't really a git-related thing so it shouldn't come from the git package, and I don't see the point of making a new package that has knowledge of git and can talk to the OS, so I'm cool with this for now.

Reply:
In this case, I think I would separate the operation where you deduce editor than the one where you actually edit the file. One is more related to config, I think it's great to solve once the whole config, and just pass around a config struct, you don't need to rely internally on os.Getenv and so on and it becomes easier to test.

Reply:
I agree